packages/discord-player/src/clientadapter/IClientAdapter.ts

```diff
@@ -0,0 +1,66 @@
+import { type Snowflake } from 'discord-api-types/globals';
+import { ChannelType } from 'discord-api-types/v10';
+import { AudioPlayerState, DiscordGatewayAdapterCreator } from "discord-voip";
+
+export const ClientType = {
+    DiscordJS: "discord.js",
+    Eris: "eris",
+    Unknown: "unknown"
+};
+
+export type ClientUser = {
+    setSelfDeaf(state?: boolean, reason?: string): Promise<void>;
```

Comment:
Can we set these state to a non undefined property?
```suggestion
    setSelfDeaf(state: boolean, reason?: string): Promise<void>;
```

Reply:
Same for other boolean logic like setSelfDeaf

Reply:
Yes but our public methods for GuildQueue.setSelfDeaf() and GuildQueue.setSelfMute() takes in mode?: boolean

I preserved this initially but as we have breaking changes anyways I will change this.

Resolved.